src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -349,10 +351,16 @@ export class EdgeProvider extends Bridgeable {
     console.log(`requestSpendUri ${uri}`)
     const { currencyWallets } = this._state.core.account
     const edgeWallet = currencyWallets[this._state.ui.wallets.selectedWalletId]
-    const result = await edgeWallet.parseUri(uri)
 
+    const result: EdgeParsedUri & { paymentProtocolURL?: string } = await edgeWallet.parseUri(uri)
     const { currencyCode = result.currencyCode, customNetworkFee, metadata, lockInputs = true, uniqueIdentifier, orderId } = options
 
+    // Check is PaymentProtocolUri
+    if (!!result.paymentProtocolURL && !result.publicAddress) {
```

Comment:
Don't use a sketchy-boolean, instead explicitly check for null like this:
`if (result.paymentProtocolURL != null && result.publicAddress == null)`

Reply:
This was copy/pasted from the logic in the AddressTile

https://github.com/EdgeApp/edge-react-gui/blob/d8788689a2b6d46fd277750543eece24c64a287b/src/components/tiles/AddressTile.js#L113

Reply:
Done. I also removed the check for `result.publicAddress == null` since that's not necessary to proceed w/payment protocol. The address existing can just be a backup but we can prefer the payment proto.